hack/chains/kaniko-cosign-verify.sh

```diff
@@ -11,11 +11,18 @@ TASKRUN_NAME=taskrun/$( trim-name $TASKRUN_NAME )
 IMAGE_URL=$( oc get $TASKRUN_NAME -o json | jq -r '.status.taskResults[1].value' )
 IMAGE_REGISTRY=$( echo $IMAGE_URL | cut -d/ -f1 )
 #IMAGE_REGISTRY=$( oc registry info )
+INTERNAL_REGISTRY=$([[ $IMAGE_REGISTRY == 'image-registry.openshift-image-registry.svc:5000' ]] && echo true || echo false)
+# From https://github.com/redhat-appstudio/build-definitions/ two images are created ${GIT_SHA}-1 and ${GIT_SHA}-2
+# the ${GIT_SHA}-1 contains the tasks we need here, there is a race condition here, the image might not have been
+# built for the HEAD ref used here, in that case wait few minutes for it to be built
+APPSTUDIO_TASKS_IMGREF=quay.io/redhat-appstudio/appstudio-tasks:$(git ls-remote --heads https://github.com/redhat-appstudio/build-definitions.git refs/heads/main|cut -f 1)-1
```

Comment:
I want to check that I understand the motivation here.

Most of the similar code in this repo seems to hard code a digest, which I guess means choosing stability in the stability/freshness tradeoff. Is that right? And are you choosing freshness over stability here intentionally? And is the reason we can't just use "latest" because there is no latest tag in this repo?

Reply:
Yeah, my motivation was that when looking at this I could not find any references to `quay.io/redhat-appstudio/appstudio-tasks` (only to `quay.io/redhat-appstudio/build-templates-bundle`), and that I like issues to be seen early. In this case we already saw that the build of the image failed (and was fixed) and in the future we could see issues with the tasks themselves.